service/history/queues/executable.go

```diff
@@ -521,10 +515,8 @@ func (e *executableImpl) HandleErr(err error) (retErr error) {
 	e.logger.Warn("Fail to process task", tag.Error(err), tag.ErrorType(err), tag.UnexpectedErrorAttempts(int32(e.unexpectedErrorAttempts)), tag.LifeCycleProcessingFailed)
 
 	if e.isUnexpectedNonRetryableError(err) {
-		// Terminal errors are likely due to data corruption.
 		// Drop the task by returning nil so that task will be marked as completed,
 		// or send it to the DLQ if that is enabled.
-		metrics.TaskCorruptionCounter.With(e.taggedMetricsHandler).Record(1)
```

Comment:
Did you intend to remove this metric increment as well? Is this not used in any alerts?

Reply:
I moved it to L444.

Re. alert, DLQ is always enabled so we will get alert when tasks get dlqed. I actually plan to remove the alert on TaskCorruptionCounter. Right now we always get two alerts...one for this corruption metric, one for dlq.

Reply:
DLQ gets more and more power. I like it.